### DISCUSSION TITLE
Image should allow different placeholder for error and loading

### DISCUSSION THREAD

**Discussion by ftzi** [mantinedev/mantine#2923] [Feature requests]
Image considers `(!loaded || error)` the same to apply the placeholder, and the placeholder will be the same component for both situations.

https://github.com/mantinedev/mantine/blob/349849916a607f34c872bd23d54786559fb3ab69/src/mantine-core/src/Image/Image.tsx#L83

https://github.com/mantinedev/mantine/blob/349849916a607f34c872bd23d54786559fb3ab69/src/mantine-core/src/Image/Image.tsx#L111-L113

The thing is: I only want to set the placeholder if there is an error, to display a specific component if it happens. When it's loading, the blank component is ok.

It's possible to have my own logic for it by using onLoad/onError and withPlaceholder/placeholder props, but I feel that the component could just allow this behavior by default.

**Comment by rtivital** [comment]
This case is only applied when you change `src` prop of the image after the initial image was loaded. Otherwise, the placeholder won't be rendered.

**Reply by ftzi** [reply]
Yes, and this is an issue for me as my image URL changes due to some other reasons.

**Reply by ftzi** [reply]
withPlaceholder could also be a function with the args ({error, loading}), so we could use those values from it without handling the state by ourselves

**Reply by rtivital** [reply]
Okay, I'll just remove the placeholder for loading state then, as it also may cause issues with ssr

**Reply by ftzi** [reply]
Thanks!

**Reply by ftzi** [reply]
Also, JFYI:

In the docs (https://mantine.dev/core/image/#placeholder), it says that it basically is required the width and height to center the placeholder (and it's, else it will stay at the top, not center), but I managed to successfully center the placeholder with 

```tsx
<Image
	[...]
	styles={{
	  imageWrapper: {
	    position: 'static', // This vertically center the placeholder
	  },
	}}
/>
```

My image doesn't have width or height and the placeholder is successfully being centered due to this `'static'`. I am not sure how this works but it does lol. 

It may be useful to you/this lib, not sure if this would be bad for some specific situation.
